core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocalEntityManager.java

```diff
@@ -742,7 +742,7 @@ private boolean unmanageNonRecursive(Entity e) {
             if (e instanceof Group) {
                 Collection<Entity> members = ((Group)e).getMembers();
                 for (Entity member : members) {
-                    if (!Entities.isNoLongerManaged(member)) member.removeGroup((Group)e);
+                    if (!Entities.isNoLongerManaged(member)) ((EntityInternal)member).groups().remove((Group)e);
```

Comment:
You're right that this should be in its own PR - It's unrelated to the private fields. I was just being lazy. I'll separate it out.

The change is just to get rid of deprecation warnings - to remove the calls to `removeGroup` and `addGroup` with the equivalent recommended calls.